### ISSUE TITLE
Orca install issue

### ISSUE THREAD

**Issue by Nandinihpc** [LLNL/spack#22804]
I was trying to do spack install orca. It needs manual download of package. After downloading when i try to install, i found the package.py file has different name to be used  for downloaded file as 
 orca_4_2_1_linux_x86-64_openmpi3.1.4.tar.xz  instead of 
orca_4_2_1_linux_x86-64_openmpi314.tar.xz
 When i renamed it and the install seemed to move to next stage and got a new error with checksum like this below.

 Installing orca-4.2.1-2ehxs6lu2klhyj6p4epruilvay2sdf66
==> No binary for orca-4.2.1-2ehxs6lu2klhyj6p4epruilvay2sdf66 found: installing from source
==> Warning: Missing a source id for slurm@20.11.0
==> Fetching file:///home/dianariv/orca_4_2_1_linux_x86-64_openmpi3.1.4.tar.xz
############################################################################################# 100.0%
==> Error: ChecksumError: sha256 checksum failed for /tmp/dianariv/spack-stage/spack-stage-orca-4.2.1-2ehxs6lu2klhyj6p4epruilvay2sdf66/orca_4_2_1_linux_x86-64_openmpi3.1.4.tar.xz
    Expected 9bbb3bfdca8220b417ee898b27b2885508d8c82799adfa63dde9e72eab49a6b2 but got a1ff07bb01ac81ce9a4d6637fef77c12d0ec45354cebc72245c4e0d1620af956

/opt/apps/nfs/spack-quanah/spack/lib/spack/spack/package.py:1364, in do_fetch:
       1361        self.stage.fetch(mirror_only, err_msg=err_msg)
       1362        self._fetch_time = time.time() - start_time
       1363
  >>   1364        if checksum and self.version in self.versions:
       1365            self.stage.check()
       1366
       1367        self.stage.cache_local()

==> Error: orca-4.2.1-2ehxs6lu2klhyj6p4epruilvay2sdf66: ChecksumError: sha256 checksum failed for /tmp/dianariv/spack-stage/spack-stage-orca-4.2.1-2ehxs6lu2klhyj6p4epruilvay2sdf66/orca_4_2_1_linux_x86-64_openmpi3.1.4.tar.xz
    Expected 9bbb3bfdca8220b417ee898b27b2885508d8c82799adfa63dde9e72eab49a6b2 but got a1ff07bb01ac81ce9a4d6637fef77c12d0ec45354cebc72245c4e0d1620af956
==> Error: orca-4.2.1-2ehxs6lu2klhyj6p4epruilvay2sdf66: Package was not installed

**Comment by tillhanke** [comment]
@snehring I also found some problems while trying to install orca. Mainly that the default version 5.0.3-f.1 is not listed in the forum. Where might one be able to find that?
Would it maybe be better to set 5.0.3 as default?
For 4.2.1 the problem still persists, that the files are different in the forum from the listed files in the spack package.
Thanks for your help.

**Comment by tillhanke** [comment]
@Nandinihpc If you are okay with the version 5.0.3 you can install it by using `spack install orca@=5.0.3` the tar file for that version has the correct hash.
If otherwise you want to add the older version you probably can just edit the package via `spack edit orca`
and replace the sha256 line 
```python
    version(
        "4.2.1",
        sha256="9bbb3bfdca8220b417ee898b27b2885508d8c82799adfa63dde9e72eab49a6b2",
        expand=False,
    )
```
with the correct hash, which you can calculate using `sha256sum`.
Please also make sure the sha1 hash of your file is the same as listed in the forum.

**Comment by snehring** [comment]
Looks like they've modified the files since the hashes were collected. I'll examine them when I get a chance. Worth pointing out though that the file for 4.2.1 should be the zst, not the xz archive.

**Comment by snehring** [comment]
The problem with the initial issue (which I am just now noticing is nearly three years old) is the use of the xz archive instead of the zst archive as specified. The checksum seems fine, despite the modification date on the forum being newer. @tillhanke your complaint with the -f version should be resolved by the PR you found. I can put a separate PR in for the 5.0.4 version once that PR is merged.
